src/main/webapp/js/instructorFeedbackEdit/customFeedbackPaths.js

```diff
@@ -0,0 +1,358 @@
+var FEEDBACK_PARTICIPANT_TYPE_CUSTOM = 'CUSTOM';
+var FEEDBACK_PARTICIPANT_TYPE_SELF = 'SELF';
+var FEEDBACK_PARTICIPANT_TYPE_STUDENTS = 'STUDENTS';
+var FEEDBACK_PARTICIPANT_TYPE_INSTRUCTORS = 'INSTRUCTORS';
+var FEEDBACK_PARTICIPANT_TYPE_TEAMS = 'TEAMS';
+var FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM = 'OWN_TEAM';
+var FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM_MEMBERS = 'OWN_TEAM_MEMBERS';
+var FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM_MEMBERS_INCLUDING_SELF = 'OWN_TEAM_MEMBERS_INCLUDING_SELF';
+var FEEDBACK_PARTICIPANT_TYPE_NONE = 'NONE';
+var TEAM_NAME_INSTRUCTORS = 'Instructors';
+
+var sessionCreator;
+var studentEmailToTeamNameMap;
+var instructorEmails;
+var teamNameToStudentEmailsMap;
+var studentEmails;
+var teamNames;
+var allPossibleFeedbackGivers;
+var allPossibleFeedbackRecipients;
+
+$(document).ready(function() {
+    initialiseCustomFeedbackPathsData();
+    initialiseFeedbackPathsSpreadsheets();
+    bindEventHandlers();
+});
+
+function initialiseCustomFeedbackPathsData() {
+    sessionCreator = $('#session-creator-data').data('session-creator');
+    studentEmailToTeamNameMap = $('#students-data').data('students');
+    instructorEmails = $('#instructors-data').data('instructors');
+    teamNameToStudentEmailsMap = [[]];
+    studentEmails = [];
+    teamNames = [];
+    
+    for (var studentEmail in studentEmailToTeamNameMap) {
+        if (studentEmailToTeamNameMap.hasOwnProperty(studentEmail)) {
+            studentEmails.push(studentEmail);
+            
+            var teamName = studentEmailToTeamNameMap[studentEmail];
+            if (!teamNames.includes(teamName)) {
+                teamNames.push(teamName);
+            }
+            
+            var studentEmailsList = teamNameToStudentEmailsMap[teamName];
+            if (studentEmailsList === undefined) {
+                studentEmailsList = [];
+            }
+            studentEmailsList.push(studentEmail);
+            teamNameToStudentEmailsMap[teamName] = studentEmailsList;
+        }
+    }
+    
+    allPossibleFeedbackGivers = [''];
+    allPossibleFeedbackGivers = allPossibleFeedbackGivers.concat(studentEmails);
+    for (var i = 0; i < instructorEmails.length; i++) {
+        if (!allPossibleFeedbackGivers.includes(instructorEmails[i])) {
+            allPossibleFeedbackGivers.push(instructorEmails[i]);
+        }
+    }
+    allPossibleFeedbackGivers = allPossibleFeedbackGivers.concat(teamNames);
+    allPossibleFeedbackRecipients = allPossibleFeedbackGivers.slice();
```

Comment:
Yup `slice()` will create a deep copy of the array as it contains only primitives. If the array contained objects, it will create a shallow copy. (However in this case, as discussed, it will not matter if it is deep or shallow as `allPossibleFeedbackRecipients` and `allPossibleFeedbackGivers` are not meant to be changed after they have been initialised)